clang/test/SemaCXX/cxx2b-deducing-this.cpp

```diff
@@ -5,7 +5,7 @@
 void f(this); // expected-error{{variable has incomplete type 'void'}} \
               // expected-error{{invalid use of 'this' outside of a non-static member function}}
 
-void g(this auto); // expected-error{{an explicit object parameter cannot appear in a non-member function}}
+void g(this auto); // expected-error{{an explicit object parameter is not allowed here}}
```

Comment:
I would rather keep the older diagnostic, which is more help. I'd recommend using %select. there are 2 cases

- In a non-member function
- In the declaration of a function pointer type

Reply:
I’d also prefer keeping it but I wasn’t entirely sure how to go about that; I can try and figure something out based on the declarator context.

Reply:
Done.